### PR TITLE
the mega pull

### PR DESCRIPTION
**Pull request by Raboo**
Hi, I'm back!

Took your feedback and did some changes.

I just merged your master into my fork and lets do the second attempt at a pull request.
Okay, the changes:
* support for MySQL/MariaDB for debian/ubuntu, added cases for RPM based distros aswell, but package names must be completed..
* bumped version a couple of steps, since its a big change.
* changed how the search works to support wrapper cookbooks.
* removed stuff that was not pertinet to this cookbook ( e.g. iptables, should belong in wrapper).
* fixed a bunch of minor stuff, (like adding not_if constraints to stop resources from executing every chef run.)

We are currently using a wrapper cookbook that includes this one and it works in Ubuntu precise. We've installed the agent in precise and trusty. Haven't tested ambari server on trusty yet, but can't imagine that it will be a problem.

### CODE REVIEW

**Comment by Joseph-R** [comment]
Hey @Raboo , 

I'm sorry this PR has lingered, man.  I'm not sure how many people are actively maintaining this project atm, and I have been more or less "hands off" last month or so.

tl;dr - I'm just getting to this now.  And it's a big PR.  I will look it over this week and get back to you.

~J

**Comment by Raboo** [comment]
Okay..

**Comment by jp** [comment]
This was merged with #24 splitting it in multiple commits helped a lot.
Thanks a lot @Raboo
